CONTRIBUTING.md

```diff
@@ -51,6 +51,44 @@ Work on your self-assigned issue and eventually create a Pull Request.
    pre-commit install
    ```
 
+1. On MacOS with Apple Silicon, if this installation approach runs into errors, you may need to install the following prerequisites:
+
+   ```bash
+   brew install cmake libomp
+   ```
+
+which require `homebrew` to be installed on your development machine.  Please install these libraries and then repeat the previous step:
+
+```bash
+pip install -e .
+```
+
+1. If the installation still does not complete successfully, then it is possible that the order of the
+   "requirements" files matters.  To carry out the installation in the proper sequence, please execute:
+
+   ```bash
+   pip install --use-pep517 --requirement requirements.txt --requirement requirements_benchmarking.txt \
+      --requirement requirements_distributed.txt --requirement requirements_explain.txt \
+      --requirement requirements_hyperopt.txt --requirement requirements_llm.txt \
+      --requirement requirements_serve.txt --requirement requirements_test.txt \
+      --requirement requirements_tree.txt --requirement requirements_viz.txt \
+      --requirement requirements_extra.txt --editable .
```

Comment:
not sure, but this should be equivalent to  pip install --use-pep517 -e .[test] (a parte fro mthe extras)
not sure what the pep directive changes, but i would verify before suggesting to manually change the requirement installation.

Also, in general, I suggest to checkout the setup.py script as it makes clear both the installation sub-packages and the order of the installation. if there are discoveries that come from reading it, I'd document them here too

Reply:
@w4nderlust Yes -- per my previous reply, I have finally realized what was going on -- you are correct (and I will fix the relevant parts, if any, in the next commit).  Thanks.